### PR TITLE
Improve Error Handling

### LINKED ISSUE #3: feat: handle improperly encoded JWT with helpful message
Errors such as JWT encoded incorrectly should be bubbled up to the user in a usable way.

JWT specifies an encoding of base64url. I have seen issuers which encode JWT as base64. Systems such as Istio expect tokens which adhere to the RFC and will fail to handle improperly encoded JWT. This tool should be able to identify such a toekn and notify the user that the encoding is non-standard.

### PR DESCRIPTION
**Pull request by ilrudie**
- introduce error handling
- switch to json formatted output for JWT
- errors are not in JSON at this time
- Makefile enhancements
- _some_ basic unit testing

fixes #3

### CODE REVIEW

File: pkg/jwt/decode.go [hunk 1]
```diff
@@ -8,58 +8,98 @@ import (
 	"time"
 )
 
+type Jwt struct {
+	Header  *jsonData `json:"header,omitempty"`
+	Payload *jsonData `json:"payload,omitempty"`
+}
+
+// Convert a JWT to a string or give "" on marhal failure
```
Comment (xytian315) [inline pkg/jwt/decode.go]: nitpick: typo marshal

File: pkg/jwt/decode.go [hunk 2]
```diff
@@ -8,58 +8,98 @@ import (
...
+	}
+	return fmt.Errorf("%s; %s", e1.Error(), e2.Error())
+}
+
+func prefixError(e error, p string) error {
+	if e != nil {
+		return fmt.Errorf("%s %s", p, e.Error())
+	}
+	return e
+}
```
Comment (xytian315) [inline pkg/jwt/decode.go]: suggestion: maybe move these two generic functions into another package/go file?

File: pkg/jwt/decode_test.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	InputVal     string
+	ExpectVal    string
+	ExpectErr    bool
+	ExpectErrVal string
+}
+
+func TestDecode(t *testing.T) {
+
+	table := []DecodeTestInput{
+		{
```
Comment (xytian315) [inline pkg/jwt/decode_test.go:20]: suggestions:
Do we want to add Description into test cases? Or is it gonna accomplish in another task?

File: pkg/jwt/decode.go (hunk 2 above)
Comment (ilrudie) [reply pkg/jwt/decode.go]: can do

File: pkg/jwt/decode_test.go (hunk 3 above)
Comment (ilrudie) [reply pkg/jwt/decode_test.go:20]: I actually just wrote these to help me with the development but I didn't want to impose a testing methodology or assertion engine on the project unilaterally. I think we should agree as a group on that first and then we can adjust these tests to fit with that decision as needed. Could be part of this PR or just moved to a different one.

**Review by xytian315 — APPROVED** [review]
🚀

**Pull request merged**